ldms/src/sampler/slurm/slurm_sampler.c

```diff
@@ -871,6 +920,15 @@ static int slurm_recv_cb(ldmsd_stream_client_t c, void *ctxt,
 			}
 			handle_job_init(job, entity);
 		}
+	} else if (0 == strncmp(event_name->str, "step_init", 9)) {
```

Comment:
This looks good. Don't we also need a new handler for "step_exit"?

Reply:
I don't think so because the only purpose of "step_init" is to update nnodes, and ntasks; however, what it won't do is update these back to zero between steps. If that is an issue, we can add that event.

Reply:
Well, this is an area where the mismatch between slurm's model of a "job" and ldms's model of a "job" causes confusion.

I should think that since the ldms slurm sampler is just calling tasks started by slurm part of a "job", that each time there is a step_init event, the number of tasks should be incremented accordingly, and during step_exit events it should decrement accordingly.

So using an interleaved example of job steps we might have:

T0 step_init, jobid A, stepid X, 3 tasks
T100 step_init, jobid A, stepid Y, 2 tasks
T200 step_fini, jobid A, stepid X
T300 step_fini, jobid A, stepid Y

So I would expect ldms to show these tasks counts at the following times:

T50 3 tasks
T150 5 tasks
T250 2 tasks
T300 0 tasks

The same _cannot_ be done for nnodes though. Job steps do not necessarily run on the same exact subset of nodes as other job steps. For instance, one job step might run on node opal[1-50], and second job step in the same allocation might run on opal[25-75]. And the encompassing job allocation for those job sets might be opal[1-1000].

Reply:
I should note that I'm assuming task count on a particular node, not task count in the entire job step.